Fix: Custom Order when using `Custom language texts` not working in a special case

mvscheidt opened the pull request:
We noticed that the custom order based on the order of `Custom language texts` does not work if you use something like `de-CH` or `en-US`

If you leave out the country code then it works (no change has been done on that).

The Reason this fails is because the method `getNormalizedLanguage()` returns something like `de_CH` or `en_US` - which leads to array_search not finding the entry in `$languages` and thus returning false even though it shouldn't.

I've choosen the current way of doing it because I didn't want to change the method `getNormalizedLanguage()` itself because it is also used elsewhere (when setting some css classes for the frontend which could cause problems for people who rely on this for something).

Edit: Notice I am not talking about it being case-sensitive - the code does strtolower anyways - the problem is dash `-` (In Contao) vs underscore `_` (from `getNormalizedLanguage()`)

aschempp commented:
Shouldn't we simply replace `getNormalizedLanguage` with `getLanguageTag`?

Also: can you add a unit test for the case?

mvscheidt commented:
Ah yeah, totally missed `getLanguageTag` - sure we should use that - no point in adding extra code when theres no need for it.

I'll take care of the test aswell.

mvscheidt commented:
Sorry for taking so long.

I've rebased / squashed the commits into one.

I've also added the unittest for the issue and ran the whole unittest suite locally (passed for me).

Overall code coverage is down though because I only created unittests which test for the bugfix and not for the rest of the classes

* https://coveralls.io/builds/12636362
* https://coveralls.io/builds/13127609

aschempp commented:
Fixed in release 3.1.7
